### PR TITLE
[v9] Backport: "helm: Add details on AWS ACM to AWS guide (#10857)"

### PR DESCRIPTION
**Pull request by webvictim**
Backports #10857 to `branch/v9`

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

File: docs/pages/kubernetes-access/helm/guides/aws.mdx [hunk 1]
```diff
@@ -134,6 +136,30 @@ You'll need to replace these values in the policy example below:
 }
 ```
 
+## Step 4. TLS certificates for Teleport
```
Comment (ptgott) [inline docs/pages/kubernetes-access/helm/guides/aws.mdx:139]:
```suggestion
## Step 4/7. TLS certificates for Teleport
```
Per the style guide

File: docs/pages/kubernetes-access/helm/guides/aws.mdx [hunk 2]
```diff
@@ -134,6 +136,30 @@ You'll need to replace these values in the policy example below:
...
+
+The `teleport-cluster` chart deploys a Kubernetes `LoadBalancer` to handle incoming connections to the Teleport proxy server.
```
Comment (ptgott) [inline docs/pages/kubernetes-access/helm/guides/aws.mdx:141]:
```suggestion
The `teleport-cluster` chart deploys a Kubernetes `LoadBalancer` to handle incoming connections to the Teleport Proxy Service.
```

File: docs/pages/kubernetes-access/helm/guides/aws.mdx [hunk 3]
```diff
@@ -228,7 +248,57 @@ $ kubectl create namespace teleport
 $ kubectl --namespace teleport create -f aws-issuer.yaml
 ```
 
-## Step 5/7. Set values to configure the cluster
+## Step 4b. Configure Teleport to use ACM to handle TLS
+
+<Admonition type="warning" title="Command-line Teleport Application access does not work with ACM">
```
Comment (ptgott) [inline docs/pages/kubernetes-access/helm/guides/aws.mdx:253]:
```suggestion
<Admonition type="warning" title="Command-line Teleport Application Access does not work with ACM">
```

File: docs/pages/kubernetes-access/helm/guides/aws.mdx [hunk 4]
```diff
@@ -228,7 +248,57 @@ $ kubectl create namespace teleport
...
+  Using ACM will prevent Teleport from handling application access via CLI (using client certificates), as Teleport
```
Comment (ptgott) [inline docs/pages/kubernetes-access/helm/guides/aws.mdx:254]:
```suggestion
  Using ACM will prevent Teleport from handling Application Access via CLI (using client certificates), as Teleport
```

File: docs/pages/kubernetes-access/helm/guides/aws.mdx [hunk 5]
```diff
@@ -228,7 +248,57 @@ $ kubectl create namespace teleport
...
+  will not be handling its own TLS termination.
+
+  If you need to use Teleport application access from the command line, you should use `cert-manager` instead (as described in
```
Comment (ptgott) [inline docs/pages/kubernetes-access/helm/guides/aws.mdx:257]:
```suggestion
  If you need to use Teleport Application Access from the command line, you should use `cert-manager` instead (as described in
```

File: docs/pages/kubernetes-access/helm/guides/aws.mdx [hunk 6]
```diff
@@ -228,7 +248,57 @@ $ kubectl create namespace teleport
...
+  To use an internal AWS network load balancer (as opposed to the default internet-facing NLB), you should add two annotations:
+
+  ```yaml
+    service.beta.kubernetes.io/aws-load-balancer-scheme: "internal"
+    service.beta.kubernetes.io/aws-load-balancer-internal: "true"
+  ```
+</Admonition>
+
+
+## Step 5. Set values to configure the cluster
```
Comment (ptgott) [inline docs/pages/kubernetes-access/helm/guides/aws.mdx:301]:
```suggestion
## Step 5/7. Set values to configure the cluster
```
Per the style guide

File: docs/pages/kubernetes-access/helm/reference.mdx [hunk 7]
```diff
@@ -539,7 +539,7 @@ cluster deployed in HA mode.
   You must install and configure `cert-manager` in your Kubernetes cluster yourself.
 
   See the [cert-manager Helm install instructions](https://cert-manager.io/docs/installation/kubernetes/#option-2-install-crds-as-part-of-the-helm-release)
-  and the relevant sections of the [AWS](./guides/aws.mdx#step-47-install-and-configure-cert-manager) and [GCP](./guides/gcp.mdx#step-47-install-and-configure-cert-manager) guides for more information.
+  and the relevant sections of the [AWS](./guides/aws.mdx#step-4-tls-certificates-for-teleport) and [GCP](./guides/gcp.mdx#step-47-install-and-configure-cert-manager) guides for more information.
```
Comment (ptgott) [inline docs/pages/kubernetes-access/helm/reference.mdx:542]:
```suggestion
  and the relevant sections of the [AWS](./guides/aws.mdx#step-47-tls-certificates-for-teleport) and [GCP](./guides/gcp.mdx#step-47-install-and-configure-cert-manager) guides for more information.
```

File: docs/pages/kubernetes-access/helm/reference.mdx [hunk 8]
```diff
@@ -554,7 +554,7 @@ Setting `highAvailability.certManager.addCommonName` to `true` will instruct `ce
...
```
Comment (ptgott) [inline docs/pages/kubernetes-access/helm/reference.mdx:557]:
```suggestion
  and the relevant sections of the [AWS](./guides/aws.mdx#step-47-tls-certificates-for-teleport) and [GCP](./guides/gcp.mdx#step-47-install-and-configure-cert-manager) guides for more information.
```

File: docs/pages/kubernetes-access/helm/reference.mdx [hunk 9]
```diff
@@ -588,7 +588,7 @@ Sets the name of the `cert-manager` `Issuer` or `ClusterIssuer` to use for issui
   You must install configure an appropriate `Issuer` supporting a DNS01 challenge yourself.
 
   Please see the [cert-manager DNS01 docs](https://cert-manager.io/docs/configuration/acme/dns01/#supported-dns01-providers) and the relevant sections
-  of the [AWS](./guides/aws.mdx#step-47-install-and-configure-cert-manager) and [GCP](./guides/gcp.mdx#step-47-install-and-configure-cert-manager) guides for more information.
+  of the [AWS](./guides/aws.mdx#step-4-tls-certificates-for-teleport) and [GCP](./guides/gcp.mdx#step-47-install-and-configure-cert-manager) guides for more information.
```
Comment (ptgott) [inline docs/pages/kubernetes-access/helm/reference.mdx:591]:
```suggestion
  of the [AWS](./guides/aws.mdx#step-47-tls-certificates-for-teleport) and [GCP](./guides/gcp.mdx#step-47-install-and-configure-cert-manager) guides for more information.
```

**Comment by ptgott** [comment]
I know this is a backport, but this commit introduces a few minor style issues to `branch/v9` that I'd like to prevent if possible.

**Comment by webvictim** [comment]
No problem, I get it. Applied all suggestions.

**Review by probakowski — APPROVED** [review]

**Pull request merged**